neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/query/parsing/impl/Neo4jQueryRendererDelegate.java

```diff
@@ -71,6 +76,35 @@ private void match(StringBuilder queryBuilder, String targetAlias, String label)
 		node( queryBuilder, targetAlias, label );
 	}
 
+	private void optionalMatch(StringBuilder queryBuilder, String targetAlias) {
```

Comment:
> I need to check both locations because you could refer to an embedded only in the SELECT clause.

Understood. I think in the long term the different parts of the query should be handled in the same way. I.e. there should be one object tree representing the entire query. It has been handled for MongoDB differently because we don't have a real query language there. That's why the builder there only applies to the WHERE clause while the SELECT is handled differently. I think for Neo4j this could be unified. But I think we can address this independently from the change at hand.